java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -5,175 +5,91 @@
-                            if (debug) {
-                                logger.info("Attempting to load library from classpath using " + libraryFromJar);
-                            }
-                            String tempLibraryPath = createTempFileFromResource(libraryFromJar, debug);
-                            if (debug) {
-                                logger.info("Copied resource " + libraryFromJar + " to location " + tempLibraryPath);
-                            }
-                            System.load(tempLibraryPath);
-                        }
-                    }
-                } catch (IOException e) {
-                    logger.log(Level.SEVERE, "Failed to load onnx-runtime library from jar");
-                    throw e;
-                }
-            }
-            ortApiHandle = initialiseAPIBase(ORT_API_VERSION_1);
-            loaded = true;
-        }
+  /**
+   * Loads the native C library.
+   *
+   * @throws IOException If it can't write to disk to copy out the library from the jar file.
+   */
+  static synchronized void init() throws IOException {
+    if (loaded) {
+      return;
     }
+    load("onnxruntime");
+    load("onnxruntime4j_jni");
+    ortApiHandle = initialiseAPIBase(ORT_API_VERSION_1);
+    loaded = true;
+  }
 
-    /**
-     * Copies out the named file from the class path into a temporary directory so it can be loaded
-     * by {@link System#load}.
-     * <p>
-     * The file is marked delete on exit. Throws {@link IllegalArgumentException} if the
-     * supplied path is not absolute.
-     * @param path The path to the file in the classpath.
-     * @param debugLogging If true turn on debug logging.
-     * @return The path to the extracted file on disk.
-     * @throws IOException If the file failed to read or write.
-     */
-    private static String createTempFileFromResource(String path, boolean debugLogging) throws IOException {
-        if (!path.startsWith("/")) {
-            throw new IllegalArgumentException("The path has to be absolute (start with '/').");
-        } else {
-            String[] parts = path.split("/");
-            String filename = parts.length > 1 ? parts[parts.length - 1] : null;
-            String prefix = "";
-            String suffix = null;
-            if (filename != null) {
-                parts = filename.split("\\.", 2);
-                prefix = parts[0];
-                suffix = parts.length > 1 ? "." + parts[parts.length - 1] : null;
-            }
+  /**
+   * Finds a named library in classpath resources, copies to temporary file on the filesystem to be
+   * loaded using {@link System#load}.
+   *
+   * @param library The bare name of the library.
+   * @throws IOException If the file failed to read or write.
+   */
+  private static void load(String library) throws IOException {
+    try {
+      logger.log(Level.FINE, "Attempting to load native library '" + library + "' from lib path");
+      System.load(library);
```

Comment:
This needs to be a `System.loadLibrary` call, as the library name isn't a fully qualified path.